### PR TITLE
move container runtime registries config under the image api

### PR DESCRIPTION
**Pull request by bparees**
After sleeping on it, i think i prefer this over https://github.com/openshift/api/pull/204 since the image config api already includes the CA configuration which is also consumed by the container runtime.

### CODE REVIEW

**Comment by bparees** [comment]
@mrunalp @adambkaplan @derekwaynecarr @umohnani8  ptal.. I think this might be a better approach for the new registry configuration api.

Note that I had to delete the existing registry struct to avoid declaring the same struct name in two places..  so if we do merge this, we'll have to be careful about orchestrating the downstream bumps.

File: config/v1/types_image.go [hunk 1]
```diff
@@ -34,9 +34,16 @@ type ImageSpec struct {
...
+	// should be trusted during imagestream import, pod image pull, and imageregistry
+	// pullthrough.
 	// The namespace for this config map is openshift-config.
 	AdditionalTrustedCA ConfigMapNameReference `json:"additionalTrustedCA,omitempty"`
+
+	// Registry contains configuration that determines how the container runtime
+	// should treat individual registries when accessing images for builds+pods. (e.g.
+	// whether or not to allow insecure access).  It does not contain configuration for the
+	// internal cluster registry.
+	Registry Registry `json:"registry,omitempty"`
```
Comment (derekwaynecarr) [inline config/v1/types_image.go]: in this context, would TrustedRegistrySources be a better name?

Comment (bparees) [reply config/v1/types_image.go]: Considering that one of the sub-fields is "BlockedRegistries" i'd say "Trusted" is inappropriate

I could get behind RegistrySources though.

Comment (mrunalp) [reply config/v1/types_image.go]: :+1: to RegistrySources

**Comment by umohnani8** [comment]
If we do this, the registry options will be configurable in the image CRD then right? That means we will be getting rid of the Registry CRD I added https://github.com/openshift/cluster-config-operator/blob/master/manifests/0000_05_config-operator_01_registry.crd.yaml?

**Comment by bparees** [comment]
> If we do this, the registry options will be configurable in the image CRD then right? That means we will be getting rid of the Registry CRD I added https://github.com/openshift/cluster-config-operator/blob/master/manifests/0000_05_config-operator_01_registry.crd.yaml?

yeah, that's my proposal.

**Comment by umohnani8** [comment]
Got it, sounds good to me :+1:

**Comment by mrunalp** [comment]
We need this in very soon so Urvashi still has time to land it before Friday.

**Comment by bparees** [comment]
@derekwaynecarr are you ok with this?

File: config/v1/types_image.go (hunk 1 above)
Comment (adambkaplan) [reply config/v1/types_image.go]: yes to RegistrySources as well. I think we'll want to keep AllowedRegistriesForImport separate - this governs the imagestream import behavior and has no bearing on the runtime registry _sources_.

**Comment by bparees** [comment]
(updated to use RegistrySources as the field name)

**Comment by adambkaplan** [comment]
@bparees looks like rename wasn't pushed?

File: config/v1/types_image.go (hunk 1 above)
Comment (derekwaynecarr) [reply config/v1/types_image.go]: i am fine w/ RegistrySources

**Comment by derekwaynecarr** [comment]
I am fine with this change to group under image, and `RegistrySource` is better than `Registry`

/approve

**Comment by adambkaplan** [comment]
@bparees looks like client-go was bumped, and now verify-client-go fails expecting a lister :disappointed:

**Comment by bparees** [comment]
@umohnani8 @adambkaplan ok this is landed, i'm bumping client-go here(https://github.com/openshift/client-go/pull/90) but there are no actual client changes now since it didnt' introduce a new top-level object.

@adambkaplan can you start an origin bump along w/ the changes to move the build controller to use this location?

@umohnani8 you should be good to update your operator(?) to start using this api.

**Pull request merged**